src/risk_atlas_nexus/blocks/risk_mapping/risk_mapper.py

```diff
@@ -152,25 +155,31 @@ def generate(
 
             # using a prompt without the template examples, as they maybe do not exist for all taxonomies
             prompts = [
-                self.inference_engine.prepare_prompt(
-                    prompt_template=RISK_IDENTIFICATION_TEMPLATE,
+                ZeroShotPromptBuilder(prompt_template=RISK_IDENTIFICATION_TEMPLATE).build(
```

Comment:
Hi @ingelise 
Instead of copying the code here, we can also call the API - GenericRiskDetector.detect(usecases). This is not an issue right now. Just a suggestion, so that we don't need to update this code whenever there are changes in the Risk Identification Service.

Entire code from line no 156 to 183 can be replaced.

Reply:
I see you only have existing risks. While the API needs ontology as well as the taxonomy. I think we should overload GenericRiskDetector with a new constructor for the risk mapping. Do you have any suggestions? We can take up this in the next release.

Reply:
@ingelise 
You can make the changes as I suggested or however you find is okay. I will take a look after you are done.